erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+from pypika import functions as fn
+
+from erpnext.utilities.bulk_transaction import task, update_logger
+
+
+class BulkTransactionLog(Document):
+    pass
+
+
+@frappe.whitelist()
+def retry_failing_transaction():
+    btp = frappe.qb.DocType("Bulk Transaction Log Detail")
+    data = (
+        frappe.qb.from_(btp)
+        .select(btp.transaction_name, btp.from_doctype, btp.to_doctype)
+        .distinct()
+        .where(btp.retried != 1)
+        .where(btp.transaction_status == "Failed")
+        .where(btp.date == fn.CurDate())
```

Comment:
You can only retry what failed today? Any reason for this restriction? 🤔

Reply:
This is for retrying all failing transaction log from list view; IMO
1. It should signify it will only attempt today's failed transaction. 
2. Single log should also have retry. (feature request)

Reply:
In the list view it shows all the log docs with their name set as current date. It doesn't show all the transactions at once. so the user can retry that specific date's failing transaction
<img width="1440" alt="Screenshot 2022-01-27 at 12 34 36 PM" src="https://user-images.githubusercontent.com/49878143/151308461-64d4ac7f-618b-49bc-85a8-4e3bf87bbdd0.png">

Reply:
> It doesn't show all the transactions at once. so the user can retry that specific date's failing transaction

Can they though? The query only appears to retry today's failed transactions.

Reply:
No it did not but now I have made the changes to retry that specific date's transaction